cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -1143,9 +1143,16 @@ function Start-WorkerServices {
   $kubelet_args_str = ${kube_env}['KUBELET_ARGS']
   $kubelet_args = $kubelet_args_str.Split(" ")
   Log-Output "kubelet_args from metadata: ${kubelet_args}"
+
+  # Windows nodes are renamed to a name <= 15 characters to join a AD domain.
+  # Both kubelet and kube-proxy should use instance_name.
+  # No impact if a node is not renamed as hostname and instance name are the same.
+  $instance_name = "$(Get-InstanceMetadata 'name' | Out-String)"
   $default_kubelet_args = @(`
-      "--pod-infra-container-image=${env:INFRA_CONTAINER}"
+      "--pod-infra-container-image=${env:INFRA_CONTAINER}",
+      "--hostname-override=${instance_name}"
```

Comment:
Would there ever be a reason not to use --hostname-override?

We could make the use of this flag conditional based on whether or not some key is set in the kube-env, but not sure if that level of control is needed.

Reply:
We can check if the hostname and instance name are different and apply the flag when they are different only. Do you feel this will be better?

Reply:
I assume that it wouldn't make a difference if we apply the flag when the hostname and instance name are the same.

I still have the question, would there ever be a reason not to use --hostname-override? If the answer is no, that's fine and we can leave this as is.

Reply:
If we can make all GKE generated name to be <= 15 characters then we won't need to override the hostname. This change will not be trivial and have user impact as users are already familiar with today's naming convention. So it is not something we can complete very soon.  I can't think out a reason to not use --hostname-override for now.